### PR TITLE
Organize scraped info

### PR DESCRIPTION
**Pull request by ingridel**
-> Added the organisational part of scraping the apps' web pages.
-> Added special functions to read/write to/from info files.
-> Added tests for the new functions.

### CODE REVIEW

File: src/info_files_helpers.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    :param file: file to read
+    :return: a list of the CSV rows in Dict format
+    """
+    with open(file, "r") as input_file:
+        dict_reader = DictReader(input_file)
+        apps_details = list(dict_reader)
+    return apps_details
+
+
+def log_failure(file: Path, message: str, exception: Exception) -> None:
```
Comment (phreeza) [inline src/info_files_helpers.py]: I would suggest using a "proper" logging utility here, either absl.logging, or logging from the standard library.

File: src/info_files_helpers.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            dict_reader = DictReader(info_file)
+            for row in dict_reader:
+                if all(data[field] == row[field] for field in fields_to_compare):
+                    return True
+            return False
+    except FileNotFoundError:
+        return False
+
+
+def read_list(file: Path) -> List[Dict[str, str]]:
```
Comment (phreeza) [inline src/info_files_helpers.py]: I think it would be very useful to replace this datastructure with a pandas dataframe. This has several features built in that will be useful for you, for example csv reading and writing, and making sure that all items in your List have the same keys.

File: test/app_page_scraper_test.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
 from pathlib import Path
 import pytest
-from src import PlayScrapingError, PlayAppPageScraper, AccessError
+from src import PlayScrapingError, PlayAppPageScraper, AccessError, SCRAPER_INFO_FILE_NAME
 
 ICON_HTML = """
```
Comment (phreeza) [inline test/app_page_scraper_test.py:7]: If you don't want these to be imported anywhere, I would suggest prepending them with a _

File: src/utils.py [hunk 4]
```diff
@@ -0,0 +1,7 @@
+SCRAPER_INFO_FILE_NAME = "apps"
```
Comment (phreeza) [inline src/utils.py]: Optional: if you want to make the code a bit more "google-style" you could use absl.flags to set these file names up in a configurable way.

File: src/__init__.py [hunk 5]
```diff
@@ -1,2 +1,4 @@
 from .app_page_scraper import PlayAppPageScraper
 from .betel_errors import BetelError, PlayScrapingError, AccessError
+from .utils import SCRAPER_INFO_FILE_NAME, SCRAPER_LOG_FILE_NAME, get_app_icon_name
```
Comment (phreeza) [inline src/__init__.py]: global comment:

The style guide says:  
> Use import statements for packages and modules only, not for individual classes or functions. Note that there is an explicit exemption for imports from the typing module.

so if you want to follow it, most of the imports in this project would need to be restructured. But I think it would also be OK to leave as is, perhaps consult Seb or Momo for a decision.

**Review by phreeza — COMMENTED** [review]
I mainly have some high level comments about which libraries to use. If you switch to Pandas as suggested, I can give this another pass after you refactor.

File: src/__init__.py (hunk 5 above)
Comment (phreeza) [reply src/__init__.py]: As per our discussion, perhaps this is fine for the public re-export of some things in the __init__.py (even though the style guide doesn't explicitly allow it), but the comment still applies to the other files.

File: src/info_files_helpers.py [hunk 6]
```diff
@@ -1,66 +1,38 @@
...
-        dict_writer = DictWriter(output_file, fieldnames=fieldnames)
-        if file.stat().st_size == 0:
-            dict_writer.writeheader()
-        dict_writer.writerow(data)
+    with open(file, "a+") as csv_file:
+        data.to_csv(csv_file, index=False, header=(file.stat().st_size == 0))
 
 
-def part_of_data_set(file: Path, data: Dict[str, str]) -> bool:
+def part_of_data_set(file: pathlib.Path, data: Dict[str, str]) -> bool:
```
Comment (phreeza) [inline src/info_files_helpers.py]: Should this also be a dataframe now?

File: test/info_files_helpers_test.py [hunk 7]
```diff
@@ -30,14 +32,4 @@ def test_read_list(self, test_dir):
 
         dict_list = read_list(file)
```
Comment (phreeza) [inline test/info_files_helpers_test.py]: no longer a list of dicts.

File: src/utils.py (hunk 4 above)
Comment (phreeza) [reply src/utils.py]: Not necessary in this PR, but i would suggest to discuss with Seb some more elegant way to handle these config variables, storing them in a python file like this seems odd to me.

File: src/app_page_scraper.py [hunk 8]
```diff
@@ -1,11 +1,12 @@
...
-from bs4 import BeautifulSoup
-from parmap import map as pmap
+import pathlib
+import urllib.error
+import urllib.request
+import logging
+import bs4
+import parmap
+import pandas as pd
 from .utils import get_app_icon_name, SCRAPER_INFO_FILE_NAME, SCRAPER_LOG_FILE_NAME
```
Comment (phreeza) [inline src/app_page_scraper.py]: IIUC, the style guide also suggests not to have these relative imports.

File: src/info_files_helpers.py [hunk 9]
```diff
@@ -1,66 +1,38 @@
...
+        return all([(apps_details[key] == data[key]).any() for key in data])
     except FileNotFoundError:
         return False
 
 
-def read_list(file: Path) -> List[Dict[str, str]]:
-    """Reads all the rows of a CSV file as a dictionary
-     ({column0: "value0", column1: "value1", ...}) and
-    returns a list containing them.
+def read_list(file: pathlib.Path) -> pd.DataFrame:
```
Comment (phreeza) [inline src/info_files_helpers.py]: Maybe change the name since this is no longer a list?

**Review by phreeza — CHANGES_REQUESTED** [review]

File: src/__init__.py [hunk 10]
```diff
@@ -1,2 +1,4 @@
-from .app_page_scraper import PlayAppPageScraper
-from .betel_errors import BetelError, PlayScrapingError, AccessError
+import src.app_page_scraper
```
Comment (phreeza) [inline src/__init__.py]: I think a common pattern is to name the directory that you have called `src` here the project name, so `betel`, even though it looks a bit odd because then you can have the repo called `betel` and then immediately another directory called `betel` inside that, it makes these imports look more sensible. This is more of a personal preference, so if you strongly prefer `src` i think that is also fine.

Comment (ingridel) [reply src/__init__.py]: Great suggestion! Thanks :)

**Review by phreeza — APPROVED** [review]
I find it a bit hard to follow in this interface if all changes have been made, but i think it looks like everything has been addressed?
